pkg/skaffold/runner/label/label.go

```diff
@@ -32,8 +34,180 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	patch "k8s.io/apimachinery/pkg/util/strategicpatch"
 )
 
+type objectType int
+
+// List of API Objects supported by the Skaffold Labeler
+const (
+	_ = iota
+	corev1Pod
+	appsv1Deployment
+	appsv1Beta1Deployment
+	appsv1Beta2Deployment
+	extensionsv1Beta1Deployment
+	corev1Service
+	appv1StatefulSet
+	appsv1Beta1StatefulSet
+	appsv1Beta2StatefulSet
+	extensionsv1Beta1DaemonSet
+	appsv1ReplicaSet
+	appsv1Beta2ReplicaSet
+)
+
+// converter is responsible for determining whether an object can be converted to a given type
+type converter func(runtime.Object) bool
+
+// patcher is responsible for applying a given patch to the provided object
+type patcher func(clientgo.Interface, string, string, []byte) error
+
+// objectMeta is responsible for returning a generic runtime.Object's metadata
+type objectMeta func(runtime.Object) *metav1.ObjectMeta
+
+var converters = map[objectType]converter{
```

Comment:
nit: could you collapse converter and objectmeta to one function that returns ok and objectmeta?

Reply:
yeah let me try that